components/constellation/network_listener.rs

```diff
@@ -99,7 +99,9 @@ impl NetworkListener {
                 };
 
                 match metadata.headers {
-                    Some(ref headers) if headers.contains_key(LOCATION) => {
+                    Some(ref headers)
+                        if headers.contains_key(LOCATION) && metadata.location_url.is_some() =>
```

Comment:
We should be able to remove the `headers.contains_key` check entirely. This corresponds with step 5.7.4 of https://html.spec.whatwg.org/multipage/browsing-the-web.html#process-a-navigate-fetch.

Reply:
oh! thanks for pointing to this part of spec! looks like we should check if `location_url` is some and validate its scheme, will change it!